src/Schema/Keywords/AnyOf.php

```diff
@@ -39,31 +39,29 @@ public function __construct(CebeSchema $parentSchema, int $type, BreadCrumb $bre
      *
      * @param mixed        $data
      * @param CebeSchema[] $anyOf
+     *
+     * @throws ValidationKeywordFailed
      */
     public function validate($data, array $anyOf) : void
     {
         try {
             Validator::arrayVal()->assert($anyOf);
             Validator::each(Validator::instance(CebeSchema::class))->assert($anyOf);
 
-            // Validate against all schemas
-            $matchedCount = 0;
             foreach ($anyOf as $schema) {
-                $breadCrumb      = $this->dataBreadCrumb;
-                $schemaValidator = new SchemaValidator($schema, $data, $this->validationDataType, $breadCrumb);
+                $schemaValidator = new SchemaValidator($this->validationDataType);
                 try {
-                    $schemaValidator->validate();
-                    $matchedCount++;
-                } catch (ValidationKeywordFailed $e) {
+                    $schemaValidator->validate($data, $schema, $this->dataBreadCrumb);
+
+                    return;
```

Comment:
Simplified flow. We do not really need to continue here if we match something

Reply:
Another great performance related catch!